src/cpl/nuopc/rof_import_export.F90

```diff
@@ -347,77 +387,115 @@ subroutine export_fields (gcomp, begr, endr, rc)
        first_time = .false.
     end if
 
+    ! Set tracers
+    nliq = ctl%nt_liq
+    nice = ctl%nt_ice
+
+    ! water tracers from lnd
     if ( ice_runoff )then
-       ! separate liquid and ice runoff
-       do n = begr,endr
-          rofl(n) =  ctl%direct(n,nliq) / (ctl%area(n)*0.001_r8)
-          rofi(n) =  ctl%direct(n,nfrz) / (ctl%area(n)*0.001_r8)
-          if (ctl%mask(n) >= 2) then
-             ! liquid and ice runoff are treated separately - this is what goes to the ocean
-             rofl(n) = rofl(n) + ctl%runoff(n,nliq) / (ctl%area(n)*0.001_r8)
-             rofi(n) = rofi(n) + ctl%runoff(n,nfrz) / (ctl%area(n)*0.001_r8)
+       ! separate water liquid and ice runoff
+       do nr = begr,endr
+          rof_liq(nr) = ctl%direct(nr,nliq) / (ctl%area(nr)*0.001_r8)
+          rof_ice(nr) = ctl%direct(nr,nice) / (ctl%area(nr)*0.001_r8)
+          if (ctl%mask(nr) >= 2) then
+             ! water liquid and ice runoff are treated separately - this is what goes to the ocean
+             rof_liq(nr) = rof_liq(nr) + ctl%runoff(nr,nliq) / (ctl%area(nr)*0.001_r8)
+             rof_ice(nr) = rof_ice(nr) + ctl%runoff(nr,nice) / (ctl%area(nr)*0.001_r8)
           end if
        end do
     else
-       ! liquid and ice runoff added to liquid runoff, ice runoff is zero
-       do n = begr,endr
-          rofl(n) = (ctl%direct(n,nfrz) + ctl%direct(n,nliq)) / (ctl%area(n)*0.001_r8)
-          if (ctl%mask(n) >= 2) then
-             rofl(n) = rofl(n) + (ctl%runoff(n,nfrz) + ctl%runoff(n,nliq)) / (ctl%area(n)*0.001_r8)
+       ! ice runoff added to water liquid runoff, ice runoff is zero
+       do nr = begr,endr
+          rof_liq(nr) = (ctl%direct(nr,nice) + ctl%direct(nr,nliq)) / (ctl%area(nr)*0.001_r8)
+          if (ctl%mask(nr) >= 2) then
+             rof_liq(nr) = rof_liq(nr) + (ctl%runoff(nr,nice) + ctl%runoff(nr,nliq)) / (ctl%area(nr)*0.001_r8)
           endif
-          rofi(n) = 0._r8
+          rof_ice(nr) = 0._r8
        end do
     end if
 
+    ! non-water liquid tracers from lnd to ocn
+    do nt = 1,ntracers_nonh2o
+       do nr = begr,endr
+          if (ctl%mask(nr) >= 2) then
+             rof_liq_nonh2o(nr,nt) = ctl%runoff(nr,nt+2) / (ctl%area(nr)*0.001_r8)
```

Comment:
replace `ctl%runoff(nr,nt+2)` with `ctl%runoff(nr,nice+nt)`, to be consistent with other code.

Reply:
Good suggestion. Thanks.